lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -66,17 +73,21 @@ def execute(self):
                     label.unlink()
                 except Exception:
                     self.logger.error("Failed to remove label for remote %s", tg_dir_r)
-                    errors += 1
+                    errors = 1
 
             if self.is_empty(tg_dir_r):
-                self.logger.info('All tools removed from host, "%s"', tg_dir_r.name)
+                self.logger.info(
+                    'All tools removed from group "%s" on host, "%s"',
+                    self.context.group,
+                    tg_dir_r.name,
+                )
                 try:
                     shutil.rmtree(tg_dir_r)
                 except OSError:
                     self.logger.error("Failed to remove remote directory %s", tg_dir_r)
-                    errors += 1
+                    errors = 1
 
-        return errors
+        return 0 if errors == 0 else 1
 
     def _clear_tools(self, name, remote):
```

Comment:
Should we add Type-hints here?

Reply:
We could, but I'm disinclined:  IMO, we should add type hints to new code and to existing function definitions when the signatures are changed or at any time substantial modifications are made to its body; however, adding type hints to functions which aren't otherwise being changed is a gray area -- it amounts to "change for change sake", which comes with costs and risks which are probably best avoided.

Reply:
OK, I ended up changing what this function returns for values, so I added the type hint to the signature.  🙂